exercises/07-LifecycleMethods.js.jsx

```diff
@@ -30,18 +30,41 @@ import React from 'react';
 // There are more lifecycle methods available.
 // Those three presented are commonly used.
 //
+
+// This could have done more than just console.log things
+
+
 // Extra task: Learn about componentWillUpdate. What's the difference between
 //             this and the componentDidUpdate method? Think about the possible
 //             use cases of this lifecycle method.
+// componentWillUpdate is called once it is known that a component should update. This shoud be used to prep for updated data
+// it SHOULD NOT trigger an update itself.
+// componentDidMount is called after the render method and the DOM operations can happen here.
+
+
 // Extra task: Learn about componentWillMount. How can it be useful?
 //             (Hint: Think about server-side rendering of React components)
+// componentWillMount is called before the render method and setting state here won't call a re-render. 
```

Comment:
componentWillMount does get called before render, but it will only be called once, when your component is initially added to the DOM. After the component is mounted, any additional calls to the render method won't trigger componentWillMount.